internal/goflags/flags_test.go

```diff
@@ -0,0 +1,87 @@
+package goflags
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrim(t *testing.T) {
+	for name, tc := range map[string]struct {
+		flags  CommandFlags
+		remove []string
+	}{
+		"not found": {
+			flags: CommandFlags{
+				Long:  map[string]string{"-long1": "long1val"},
+				Short: []string{"-short1"},
+			},
+			remove: []string{"-notfound"},
+		},
+		"found": {
+			flags: CommandFlags{
+				Long:  map[string]string{"-long1": "long1val", "-long2": "long2val"},
+				Short: []string{"-short1", "-short2"},
+			},
+			remove: []string{"-short1", "-long1"},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			tc.flags.Trim(tc.remove...)
+			for _, flag := range tc.remove {
+				require.NotContains(t, tc.flags.Long, flag)
+				require.NotContains(t, tc.flags.Short, flag)
+			}
+		})
+
+	}
+}
+
+func TestParse(t *testing.T) {
```

Comment:
This should also cater for the fancy arguments that pass down to the linker, etc...

- `-gcflags '-N -L'` 
- `-ldflags '-extldflags "-lm -lstdc++ -static"'`

I suspect these wouldn't work properly with the current logic...

Reply:
Added test cases for special flags